modules/video/src/videoplayer.cpp

```diff
@@ -0,0 +1,949 @@
+                _fps = 24.0;
+                LWARNING("Detected fps was 0. Falling back on 24 fps");
+                break;
+            }
+            if (!fps) {
+                LERROR("Could not find fps property");
+                break;
+            }
+            _fps = *fps;
+            if (_playbackMode == PlaybackMode::MapToSimulationTime) {
+                updateFrameDuration();
+            }
+
+            LINFO(fmt::format("Detected fps: {}", *fps));
+            _seekThreshold = 2.0 * (1.0 / _fps);
+            break;
+        }
+        case MpvKey::Pause: {
+            int* videoIsPaused = reinterpret_cast<int*>(prop->data);
+            _isPaused = *videoIsPaused == 1;
+            LINFO(fmt::format("Is Paused: {}", _isPaused));
+            break;
+        }
+        case MpvKey::Meta: {
+            LINFO("Printing meta data reply");
+            if (node.format == MPV_FORMAT_NODE_MAP) {
+                for (int n = 0; n < node.u.list->num; n++) {
+                    if (node.u.list->values[n].format == MPV_FORMAT_STRING) {
+                        LINFO(node.u.list->values[n].u.string);
+                    }
+                }
+            }
+            else {
+                LWARNING("No meta data could be read");
+            }
+
+            break;
+        }
+        case MpvKey::Params: {
+            if (node.format == MPV_FORMAT_NODE_ARRAY ||
+                node.format == MPV_FORMAT_NODE_MAP)
+            {
+                mpv_node_list* list = node.u.list;
+
+                mpv_node width, height;
+                bool foundWidth = false;
+                bool foundHeight = false;
+                for (int i = 0; i < list->num; ++i) {
+                    if (foundWidth && foundHeight) {
+                        break;
+                    }
+
+                    if (list->keys[i] == "w") {
+                        width = list->values[i];
+                        foundWidth = true;
+                    }
+                    else if (list->keys[i] == "h") {
+                        height = list->values[i];
+                        foundHeight = true;
+                    }
+                }
+
+                if (!foundWidth || !foundHeight) {
+                    LINFO("Could not find width or height params from parameters");
+                    return;
+                }
+
+                int w = -1;
+                int h = -1;
+                if (width.format == MPV_FORMAT_INT64) {
+                    w = width.u.int64;
+                }
+                if (height.format == MPV_FORMAT_INT64) {
+                    h = height.u.int64;
+                }
+
+                if (w == -1 || h == -1) {
+                    LERROR("Invalid width or height params");
+                    return;
+                }
+                resizeFBO(w, h);
+            }
+            break;
+        }
+
+        default: {
+            throw ghoul::MissingCaseException();
+            break;
```

Comment:
No need for the `break` here, since the `throw` in the line above will yeet us out of this function anyway